packages/nextjs/lib/rules/pages-need-getLink-export.ts

```diff
@@ -0,0 +1,212 @@
+
+export type Options = Record<never, never>[]
+
+export type MessageIds =
+    | "page-need-getLink-export"
+    | "no-return-statement"
+    | "no-pathname-in-return-argument"
+    | "pathname-does-not-match-page-path"
+
+export default creator<Options, MessageIds>({
+    name: "pages-need-getLink-export",
+    meta: {
+        type: "problem",
+        docs: {
+            description: "Pages must export a getLink function",
+            recommended: "error",
+        },
+        fixable: "code",
+        messages: {
+            "page-need-getLink-export":
+                "Page needs a getXYZLink function export",
+            "no-return-statement": "getLink function needs a return statement",
+            "no-pathname-in-return-argument": "No pathname in return argument",
+            "pathname-does-not-match-page-path":
+                "The returned pathname ({{pathName}}) does not match the path to the page {{pagePath}}",
+        },
+        schema: [{}],
+    },
+    defaultOptions: [],
+    create(context) {
+        let foundLinkFunction = false
+
+        return {
+            "Program:exit"(node) {
+                const path = context.getPhysicalFilename?.()
+
+                if (path == undefined) {
+                    console.error("Got no physical file name ?!")
+                    return
+                }
+
+                // no page file
+                if (!path?.includes("pages")) {
+                    return
+                }
+
+                if (!foundLinkFunction) {
+                    context.report({
+                        node,
+                        messageId: "page-need-getLink-export",
+                    })
+                }
+            },
+
+            ExportNamedDeclaration(node) {
+                const path = context.getPhysicalFilename?.()
+
+                if (path == undefined) {
+                    console.error("Got no physical file name ?!")
+                    return
+                }
+
+                if (node.type !== "ExportNamedDeclaration") {
+                    return
+                }
+
+                // no page file
+                if (!path?.includes("pages")) {
+                    return
+                }
+
+                const definition = (node as ExportNamedDeclaration).declaration
+
+                if (definition?.type !== "FunctionDeclaration") {
+                    return
+                }
+
+                if (definition.id?.type !== "Identifier") {
+                    return
+                }
+
+                // we have an incorrect name
+                if (!/get(.*?)Link/.test(definition.id.name)) {
+                    return
+                }
+
+                // we have a valid function
+                foundLinkFunction = true
+
+                const body = definition.body.body
+
+                // do we want to allow multiple returns in ifs / whiles?
+                const returnStatement = body.find(
```

Comment:
What do you mean with checking the page modules d ts?
There is no d ts in this context?

I think we can allow us to check this though, because this already is no opinionated anyway.
I don't really expect (and maybe I don't even want to ^^) these functions to be more complex than `return {pathname: "...."}`.

If we stick with the highly opinionated route, we could even check the return type on this as well. 🤔